packages/devtools_app/lib/src/inspector/inspector_tree_flutter.dart

```diff
@@ -77,17 +77,20 @@ class _InspectorTreeRowState extends State<_InspectorTreeRowWidget>
 class InspectorTreeControllerFlutter extends Object
     with InspectorTreeController, InspectorTreeFixedRowHeightController {
   /// Client the controller notifies to trigger changes to the UI.
-  InspectorControllerClient get client => _client;
-  InspectorControllerClient _client;
+  final Set<InspectorControllerClient> _clients = {};
 
-  set client(InspectorControllerClient value) {
-    if (_client == value) return;
-    // Do not set a new client if there is still an old client.
-    assert(value == null || _client == null);
-    _client = value;
+  void addClient(InspectorControllerClient value) {
```

Comment:
due to order disposed methods were called the previous checks in this method were too strict and caused issues.